src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java

```diff
@@ -0,0 +1,580 @@
+                                                                ) {
+        PublicShareDialogFragment publicShareDialogFragment = new PublicShareDialogFragment();
+        Bundle args = new Bundle();
+        args.putParcelable(ARG_FILE, fileToShare);
+        args.putParcelable(ARG_SHARE, publicShare);
+        args.putParcelable(ARG_ACCOUNT, account);
+        publicShareDialogFragment.setArguments(args);
+        return publicShareDialogFragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (getArguments() != null) {
+            mFile = getArguments().getParcelable(ARG_FILE);
+            mAccount = getArguments().getParcelable(ARG_ACCOUNT);
+            mPublicShare = getArguments().getParcelable(ARG_SHARE);
+        }
+
+        if (mFile == null && mPublicShare == null) {
+            throw new IllegalStateException("Both ARG_FILE and ARG_SHARE cannot be NULL");
+        }
+
+        setStyle(DialogFragment.STYLE_NO_TITLE, 0);
+
+        refreshCapabilitiesFromDB();
+    }
+
+    private boolean updating() {
+        return (mPublicShare != null);
+    }
+
+    private boolean isSharedFolder() {
+        return (
+            (mFile != null && mFile.isFolder()) ||
+            (mPublicShare != null && mPublicShare.isFolder())
+        );
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.share_public_dialog, container, false);
+
+        // Show or hide edit permission section
+        if (isSharedFolder()){
+            getEditPermissionSection(view).setVisibility(View.VISIBLE);
+        } else {
+            getEditPermissionSection(view).setVisibility(View.GONE);
+        }
+
+        // Show default date enforced by the server, if any
+        if (mCapabilities.getFilesSharingPublicExpireDateDays() > 0) {
+
+            getExpirationDateSwitch(view).setChecked(true);
+
+            String formattedDate = SimpleDateFormat.getDateInstance().format(
+                    DateUtils.addDaysToDate(
+                            new Date(),
+                            mCapabilities.getFilesSharingPublicExpireDateDays()
+                    )
+            );
+
+            getExpirationDateValue(view).setVisibility(View.VISIBLE);
+
+            getExpirationDateValue(view).setText(formattedDate);
+        }
+
+        // Fill in the different fields if the share is being updated
+        if (updating()) {
+
+            // Set dialog title to edit
+            getDialogTitle(view).setText(R.string.share_via_link_edit_title);
+
+            // Set existing share name
+            getNameValue(view).setText(mPublicShare.getName());
+
+            // Set the edit permissions, if any
+            if (mPublicShare.getPermissions() == CRUD_PERMISSIONS) {
```

Comment:
This comparison is not safe enough. We need to check if any of the CUD permissions is enabled.